### PR TITLE
Crowdstrike Detections

### PR DESCRIPTION
**Pull request by theycallmecharlie**
Adding a workflow of crowdstrike detections

Get the client and secret: https://www.crowdstrike.com/blog/tech-center/get-access-falcon-apis/

Use this if you have a multi tenant qradar and already a crowdstrike app installed to get events from other tenants with different log sources identifiers

### CODE REVIEW

File: Community Developed/Crowdstrike/crowdstrike.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        </CallEndpoint>
+        <Set path="/total" value="${count(/get_detections_ids/body/resources)}" />
+        <CallEndpoint url="https://${/api_host}/detects/entities/summaries/GET/v1?filter=created_timestamp:>'${/created_timestamp}'" method="POST" savePath="/get_detection_details">
+            <RequestHeader name="Accept" value="application/json" />
+            <RequestHeader name="Authorization" value="Bearer ${/bearer_token}" />
+            <RequestBody type="application/json" encoding="UTF-8">
+                {"ids":${/get_detections_ids/body/resources}}
+            </RequestBody>
+        </CallEndpoint>
+        <Set path="/total_detections" value="${count(/get_detection_details/body/resources)}" />
```
Comment (ChrisCollinsIBM) [inline Community Developed/Crowdstrike/crowdstrike.xml:30]: Do you need to rely on the `total_detections` value being correct? Or can you just iterate over the `${/get_detection_details/body/resources[]}` array?

And is `total_detections` based on the number returned by the query? Or the number on the active page? (See the paging question above)

File: Community Developed/Crowdstrike/crowdstrike.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            </UrlEncodedFormRequestBody>
+        </CallEndpoint>
+        <Set path="/bearer_token" value="${/get_access_token/body/access_token}" />
+        <FormatDate pattern="YYYY-MM-dd'T'HH:mm:ss" timeZone="UTC" time="${time() - 60 * 1000}" savePath="/created_timestamp" />
+        <CallEndpoint url="https://${/api_host}/detects/queries/detects/v1" method="GET" savePath="/get_detections_ids">
+            <RequestHeader name="Accept" value="application/json" />
+            <RequestHeader name="Authorization" value="Bearer ${/bearer_token}" />
+        </CallEndpoint>
+        <Set path="/total" value="${count(/get_detections_ids/body/resources)}" />
+        <CallEndpoint url="https://${/api_host}/detects/entities/summaries/GET/v1?filter=created_timestamp:>'${/created_timestamp}'" method="POST" savePath="/get_detection_details">
```
Comment (ChrisCollinsIBM) [inline Community Developed/Crowdstrike/crowdstrike.xml:23]: Does this API have paging? And if so how many detections will be returned until you need account for paging?

Most APIs will only return so many rows before you need to page either via a record count offset or some type of paging token.

File: Community Developed/Crowdstrike/crowdstrike.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            <RequestHeader name="Accept" value="application/json" />
+            <RequestHeader name="Authorization" value="Bearer ${/bearer_token}" />
+            <RequestBody type="application/json" encoding="UTF-8">
+                {"ids":${/get_detections_ids/body/resources}}
+            </RequestBody>
+        </CallEndpoint>
+        <Set path="/total_detections" value="${count(/get_detection_details/body/resources)}" />
+        <Set path="/i" value="0" />
+        <Set path="/j" value="${/total_detections}" />
```
Comment (ChrisCollinsIBM) [inline Community Developed/Crowdstrike/crowdstrike.xml:32]: The `j` value is only used once, you can probably just reference `${/total_detections}` again on line 34 and remove this.

**Review by ChrisCollinsIBM — CHANGES_REQUESTED** [review]
Thanks for the submission!

Overall this looks like it would be functional with a small number of detections but I have some concerns it might have issues regarding paging in general and as the `while` loop relates to total_events.

Have a look at my comments and let me know what you think.

File: Community Developed/Crowdstrike/crowdstrike.xml (hunk 3 above)
Comment (theycallmecharlie) [reply Community Developed/Crowdstrike/crowdstrike.xml:32]: Hi chris, I overwrited this structure by the foreach

File: Community Developed/Crowdstrike/crowdstrike.xml (hunk 1 above)
Comment (theycallmecharlie) [reply Community Developed/Crowdstrike/crowdstrike.xml:30]: Yes, now the total_detections is being iterated with a foreach estructure
There's no pagination on response.

**Comment by theycallmecharlie** [comment]
@ChrisCollinsIBM  i commited a update to crowdstrike.xml, what do u think?

**Review by ChrisCollinsIBM — APPROVED** [review]

**Pull request merged**